### PR TITLE
[Enh] Double-click to zoom in on the clicked location (alt to zoom out)

### LINKED ISSUE #212: DoubleClick to 2x zoom
## 🚀 Feature
As requested by @jni in #211 - it can be very nice to double click on a point to zoom 2x instead of having to scroll in. The idea was that this feature should apply whenever pan/zoom is enabled on any layer type.

### PR DESCRIPTION
**Pull request by psobolewskiPhD**
# References and relevant issues
Closes https://github.com/napari/napari/issues/212

# Description
Adds a viewer callback for double click that checks if the selected layer is in PAN_ZOOM mode.
With double click, zooms in and adjusts center to keep the point under the mouse.
With Alt, double click zooms out.

Also add a test.

### CODE REVIEW

**Comment by psobolewskiPhD** [comment]
@jni What do you think? I guess it was your feature request 😄

**Comment by DragaDoncila** [comment]
@psobolewskiPhD I pulled this down to play with it and it's neat! I updated with main in case we've fixed the setup-qtlibs thing, but ping me once this is green and ready to go!

**Comment by psobolewskiPhD** [comment]
@DragaDoncila yeah got the CI thing fixed upstream in the action, so tests should be OK here.

File: napari/components/_viewer_mouse_bindings.py [hunk 1]
```diff
@@ -13,3 +16,26 @@ def dims_scroll(viewer, event):
...
             viewer.dims._scroll_progress -= 1
+
+
+def double_click_to_zoom(viewer, event):
+    """Zoom in on double click by zoom_factor; zoom out with Alt."""
+    if (
+        viewer.layers.selection.active
+        and viewer.layers.selection.active.mode != 'pan_zoom'
+    ):
+        return
```
Comment (DragaDoncila) [inline napari/components/_viewer_mouse_bindings.py:27]: Would be good to cover this in the test as well by changing the active mode and making sure nothing happens!

**Review by DragaDoncila — COMMENTED** [review]
Looks ok to me! Made a tiny nitpick comment heh...

**Review by DragaDoncila — APPROVED** [review]
Thanks for adding that last bit of coverage @psobolewskiPhD! I think this is good to go

**Pull request merged**